packages/web/components/layouts/main.tsx

```diff
@@ -182,3 +179,52 @@ const OsmosisFullLogo: FunctionComponent<{
     }}
   />
 );
+
+interface DrawerProps {
+  children: ReactNode;
+  showSidebar: boolean;
+  isMobile: boolean;
+  setIsOpen?: (value: boolean) => void;
+}
+
+const Drawer: React.FC<DrawerProps> = ({
+  children,
+  showSidebar,
+  isMobile,
+  setIsOpen = () => console.log("close"),
+}: DrawerProps) => {
+  if (!isMobile)
+    return (
+      <article className="fixed flex flex-col inset-y-0 z-40 bg-card px-2 py-6 w-sidebar">
+        {children}
+      </article>
+    );
+
+  return (
+    <main
+      className={
+        " fixed overflow-hidden z-40 bg-backdrop inset-0 transform ease-in-out w-full h-full" +
+        (showSidebar && isMobile
+          ? " transition-opacity opacity-100 duration-300 translate-x-0 "
+          : " transition-all delay-150 opacity-0 -translate-x-full  ")
```

Comment:
Please trim all leading and trailing spaces from class definitions.

Reply:
I still see spaces in classnames.

Also, we have a classnames function that can be used to combine stateful class definitions (instead of using `+` operator).